native/src/routes/jpal-tracking/JpalTracking.js

```diff
@@ -0,0 +1,138 @@
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const errorDisplayTime = 5000
+
+const appSettings = new AppSettings()
+
+const JpalTracking = (props: PropsType) => {
+  const [settings, setSettings] = useState(defaultSettings)
+  const [settingsLoaded, setSettingsLoaded] = useState(false)
+  const [displayedTrackingCode, setDisplayedTrackingCode] = useState('')
+  const [displayedTrackingEnabled, setDisplayedTrackingEnabled] = useState(false)
+  const [error, setError] = useState(false)
+  const routeTrackingCode = props.route.params.trackingCode
+
+  const { t } = useTranslation(['settings', 'error'])
+
+  useEffect(() => {
+    if (routeTrackingCode) {
+      appSettings.setJpalTrackingCode(routeTrackingCode)
+    }
+  }, [routeTrackingCode])
+
+  useEffect(() => {
+    const loadSettings = async () => {
+      try {
+        const loadedSettings = await appSettings.loadSettings()
+        setSettingsLoaded(true)
+        setSettings(loadedSettings)
+        setDisplayedTrackingCode(loadedSettings.jpalTrackingCode)
+        setDisplayedTrackingEnabled(loadedSettings.jpalTrackingEnabled)
+        setTimeout(() => setError(false), errorDisplayTime)
+      } catch (e) {
+        setError(true)
+      }
+    }
+    loadSettings()
+  }, [])
+
+  const toggleTrackingEnabled = () => {
+    setDisplayedTrackingEnabled(!displayedTrackingEnabled)
+    appSettings.setJpalTrackingEnabled(!settings.jpalTrackingEnabled).catch(() => {
+      setError(true)
+    })
+  }
+
+  const setTrackingCode = (value: string) => {
+    setDisplayedTrackingCode(value)
+    appSettings.setJpalTrackingCode(value).catch(() => {
+      setError(true)
+    })
+  }
+
+  if (!settingsLoaded) {
+    return <LayoutContainer />
+  }
+
+  const { theme } = props
+
+  return (
+    <LayoutContainer>
+      <View style={{ padding: 40 }}>
+        <Caption title={t('tracking')} theme={theme} />
+        <Text>{t('trackingDescription')}</Text>
+
+        {error && <ErrorText>{t('error:generalError')}</ErrorText>}
+
+        <DescriptionContainer onPress={toggleTrackingEnabled}>
+          <ThemedText theme={props.theme}>{t('allowTracking')}</ThemedText>
+          <Switch
+            thumbColor={props.theme.colors.themeColor}
+            trackColor={{ true: props.theme.colors.themeColor }}
+            value={displayedTrackingEnabled}
+            onValueChange={toggleTrackingEnabled}
+            testID='switch'
+          />
+        </DescriptionContainer>
+
+        <ThemedText theme={props.theme}>{t('trackingCode')}</ThemedText>
+        <Input
+          value={displayedTrackingCode}
+          onChangeText={setTrackingCode}
+          theme={theme}
+          editable={displayedTrackingEnabled}
+          testID='input'></Input>
```

Comment:
```suggestion
        <Input
          value={displayedTrackingCode}
          onChangeText={setTrackingCode}
          theme={theme}
          editable={displayedTrackingEnabled}
          testID='input' />
```

Reply:
Could I have a grey border if it is not editable please? :stuck_out_tongue:

Reply:
It is there already :)